htlcswitch/interceptable_switch.go

```diff
@@ -129,6 +129,10 @@ type FwdResolution struct {
 	// FwdActionSettle.
 	Preimage lntypes.Preimage
 
+	// IncomingAmountMsat is the amount that is to be used for validating if
+	// Action is FwdActionResumeModified.
+	IncomingAmountMsat fn.Option[lnwire.MilliSatoshi]
```

Comment:
Yes, actually they are used in completely different scenarios. You'd likely only ever modify one. For an `asset -> BTC` forward you'd update the `IncomingAmountMsat` to tell the validation logic it's fine to accept the incoming BTC amount in the HTLC since the actual "worth" of the HTLC is `IncomingAmountMsat`.

For a `BTC -> asset` forward, you set `OutgoingAmountMsat` to tell the forwarding logic to only put `OutgoingAmountMsat` in the on-chain HTLC, even though the actual "worth" of the HTLC as a whole might be way more.